mypy/typeanal.py

```diff
@@ -305,7 +308,8 @@ def try_analyze_special_unbound_type(self, t: UnboundType, fullname: str) -> Opt
         elif fullname == 'typing.Callable':
             return self.analyze_callable_type(t)
         elif (fullname == 'typing.Type' or
-             (fullname == 'builtins.type' and self.api.is_future_flag_set('annotations'))):
+             (fullname == 'builtins.type' and self.api.is_future_flag_set('annotations')) or
```

Comment:
nit: this would read cleaner as
```
fullname == 'builtins.type' and (self.options.python_version >= (3, 9) or self.api.is_future_flag_set('annotations'))
```